node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+                service_fee_opt: Some(ServiceFeeImmoderateInsufficiency {
+                    total_service_fee_required_minor: 0,
+                    cw_service_fee_balance_minor: 0,
+                }),
+            },
+            PaymentAdjusterError::LateNotEnoughFeeForSingleTransaction {
+                original_number_of_accounts: 0,
+                number_of_accounts: 0,
+                original_service_fee_required_total_minor: 0,
+                cw_service_fee_balance_minor: 0,
+            },
+        ];
+        let inputs_count = inputs.len();
+        let results = inputs
+            .into_iter()
+            .map(|err| err.insolvency_detected())
+            .collect::<Vec<_>>();
+        assert_eq!(results, vec![true, true, true, true, true]);
+        assert_eq!(inputs_count, PaymentAdjusterError::VARIANT_COUNT + 2)
+    }
+
+    #[test]
+    fn tinier_but_larger_in_weight_account_is_prioritized_and_gains_up_to_its_disqualification_limit(
+    ) {
+        let cw_service_fee_balance_minor = multiple_by_billion(4_200_000);
+        let determine_limit_params_arc = Arc::new(Mutex::new(vec![]));
+        let mut account_1 = make_analyzed_account_by_wallet("abc");
+        let balance_1 = multiple_by_billion(3_000_000);
+        let disqualification_limit_1 = multiple_by_billion(2_300_000);
+        account_1.qualified_as.bare_account.balance_wei = balance_1;
+        account_1.disqualification_limit_minor = disqualification_limit_1;
+        let mut account_2 = make_analyzed_account_by_wallet("def");
+        let wallet_2 = account_2.qualified_as.bare_account.wallet.clone();
+        let balance_2 = multiple_by_billion(2_500_000);
+        let disqualification_limit_2 = multiple_by_billion(1_800_000);
+        account_2.qualified_as.bare_account.balance_wei = balance_2;
+        account_2.disqualification_limit_minor = disqualification_limit_2;
+        let largest_exceeding_balance = (balance_1
+            - account_1.qualified_as.payment_threshold_intercept_minor)
+            .max(balance_2 - account_2.qualified_as.payment_threshold_intercept_minor);
+        let mut subject = make_initialized_subject(
+            None,
+            Some(cw_service_fee_balance_minor),
+            None,
+            Some(largest_exceeding_balance),
+            None,
+        );
+        let disqualification_gauge = DisqualificationGaugeMock::default()
+            .determine_limit_result(disqualification_limit_2)
+            .determine_limit_result(disqualification_limit_1)
+            .determine_limit_result(disqualification_limit_1)
+            .determine_limit_params(&determine_limit_params_arc);
+        subject.disqualification_arbiter =
+            DisqualificationArbiter::new(Box::new(disqualification_gauge));
+        let weighted_payables = vec![
+            WeightedPayable::new(account_1, multiple_by_billion(2_000_100)),
+            WeightedPayable::new(account_2, multiple_by_billion(3_999_900)),
+        ];
```

Comment:
Let's create variables for these weights, so that it can be understood that it was staged.